Core/include/Acts/Propagator/DirectNavigator.hpp

```diff
@@ -171,9 +172,9 @@ class DirectNavigator {
   /// @param position The start position
   /// @param direction The start direction
   /// @param propagationDirection The propagation direction
-  void initialize(State& state, const Vector3& position,
-                  const Vector3& direction,
-                  Direction propagationDirection) const {
+  Result<void> initialize(State& state, const Vector3& position,
```

Comment:
I would suggest using `nodiscard` for all initialize calls, since it is dangerous that we silently ignore the return value here, especially, because it was a `void` function before.

Reply:
one more reason why exceptions would be better

Reply:
actually true. Now where I think of it, the whole argument of "it forces me to handle errors" breaks down in case of `void` functions...

Reply:
But this would be circumvented if `no_discard` would be the default of course...